src/Apps/Order/Components/OrderStepper.tsx

```diff
@@ -62,66 +66,57 @@ export const OrderStepper: FC<React.PropsWithChildren<OrderStepperProps>> = ({
       completedSteps.push("Review")
     }
 
-    setCompletedSteps(completedSteps)
-  }, [order])
+    return completedSteps
+  }, [order, steps, currentStep])
+
+  const stepIndex = steps.indexOf(currentStep)
 
   const handleStepClick = (step: string) => {
-    const clickedStepIndex = steps.indexOf(step)
-    setStepIndex(clickedStepIndex)
+    if (typeof window === "undefined") return
+
+    // handle counteroffer flow
+    if (completedOrderSteps.includes("Respond")) {
+      router.push(`/orders/${order.internalID}/respond`)
+      return
+    }
 
-    const activeStep = currentStep.toLocaleLowerCase()
-    const nextStep = step.toLocaleLowerCase()
-    router.push(window.location.pathname.replace(activeStep, nextStep))
+    router.push(
+      window.location.pathname.replace(
+        currentStep.toLowerCase(),
+        step.toLowerCase(),
+      ),
+    )
   }
 
+  const renderStepper = () => (
+    <Stepper
+      initialTabIndex={stepIndex}
+      currentStepIndex={stepIndex}
+      disableNavigation={false}
+    >
+      {steps.map(step => (
+        <Step
+          name={
+            completedOrderSteps.includes(step) ? (
+              <Clickable onClick={() => handleStepClick(step)}>
+                {step}
+              </Clickable>
+            ) : (
+              step
+            )
+          }
+          key={step}
+        />
+      ))}
+    </Stepper>
+  )
```

Comment:
Can this be defined as a component outside of the body instead of a `render_()` function that returns JSX like a component?